scripts/rax_create_image.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python2
+#
+# Copyright 2016, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Basic tool to save an image of a server in Rackspace Cloud. While
+   in general it is preferred to use the nova/openstack CLI to do this,
+   in our Jenkins jobs we only have access to pyrax and the nova CLI
+   that comes with it which does not work. Until we move to using the
+   shade-based Ansible modules this little tool will be necessary."""
+
+import argparse
+import pyrax
+
+
+def main():
+    """Run the main application."""
+
+    # Setup argument parsing
+    parser = argparse.ArgumentParser(
+        description='Basic pyrax CLI utilities',
+        epilog='Licensed "Apache 2.0"')
+
+    parser.add_argument(
+        '-c',
+        '--credentialsfile',
+        help='<Required> The path to the pyrax.cfg file',
+        default='~/.pyrax.cfg'
+    )
+
+    parser.add_argument(
+        '-r',
+        '--cloudregion',
+        help='<Required> The region to execute commands against',
+        required=True
+    )
+
+    parser.add_argument(
+        '-s',
+        '--serveruuid',
+        help='<Required> The server uuid to execute commands against',
+        required=True
+    )
+
+    parser.add_argument(
+        '-i',
+        '--imagename',
+        help='<Required> The name to use for the saved image',
+        required=True
+    )
+
+    # Parse arguments
+    args = parser.parse_args()
+
```

Comment:
nit: I suppose here we assume arguments are always good and well introduced? That is because this cli will only be used in gating, where we own the arguments given into this cli?

Reply:
Yes. The hope is that we phase out the use of this script altogether and shift to using the openstack CLI very soon instead - but for now pyrax has to be used.